libgalois/src/PropertyGraph.cpp

```diff
@@ -1027,23 +1027,23 @@ katana::CreateSymmetricGraph(katana::PropertyGraph* pg) {
 }
 
 katana::Result<std::unique_ptr<katana::PropertyGraph>>
-katana::CreateTransposeGraph(katana::PropertyGraph* pg) {
+katana::CreateTransposeGraphTopology(katana::PropertyGraph* pg) {
   const GraphTopology& topology = pg->topology();
   auto transpose = std::make_unique<katana::PropertyGraph>();
   if (topology.num_nodes() == 0) {
     return std::unique_ptr<PropertyGraph>(std::move(transpose));
   }
 
   // New transpose graph topology
-  auto out_indices = std::make_unique<katana::LargeArray<uint64_t>>();
-  auto out_dests = std::make_unique<katana::LargeArray<uint32_t>>();
+  katana::LargeArray<uint64_t> out_indices;
+  katana::LargeArray<uint32_t> out_dests;
```

Comment:
This should be moved out. Into the caller.

Move out the allocation and pass the `GraphTopology` as a parameter. This function should only set the values in it.

Reply:
Yes that's what I heard during the meeting. Yes I will do.